### PR TITLE
fix: Metrics with dot in name appear correctly in trial view [DET-9691]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Fixes an issue where `bun.Ident`, which we used to allow metrics names with /, was writing metric names such as `val.loss` into SQL as `AS "val"."loss"`

Example experiment with API error: http://latest-main.determined.ai:8080/det/experiments/3246/overview

The fix renames the metric for the query (i.e. `val·loss`) for the query, and restores the original name in the JSON response

## Test Plan

In an experiment such as `examples/tutorials/mnist_pytorch`, rename `validation_loss` in `const.yaml` and `model_def.py` to `val.loss`

Run the experiment and observe you can plot `val.loss` with valid API responses

Update: verify that included unit test covers the issue of metric names with a "." , "/" , or other special characters

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-9691

### CODE REVIEW

File: master/internal/api_trials_intg_test.go [hunk 1]
```diff
@@ -467,6 +467,40 @@ func TestTrialsNonNumericMetrics(t *testing.T) {
 	})
 }
 
+func TestUnusualMetricNames(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+	expectedMetricsMap := map[string]any{
+		"a.loss": 1.5,
```
Comment (NicholasBlaskey) [inline master/internal/api_trials_intg_test.go:473]: Maybe exhaustively test ascii printable characters here to make sure no other characters cause a similar issue

File: master/internal/trials/postgres_trials.go [hunk 2]
```diff
@@ -70,8 +71,8 @@ func MetricsTimeSeries(trialID int32, startTime time.Time,
 		case db.MetricTypeBool:
 			cast = "boolean"
 		}
-		subq = subq.ColumnExpr("(metrics->?->>?)::? as ?",
-			metricsObjectName, metricName, bun.Safe(cast), bun.Ident(metricName))
+		subq = subq.ColumnExpr("(metrics->?->>?)::? as ?", metricsObjectName,
+			metricName, bun.Safe(cast), bun.Ident(strings.ReplaceAll(metricName, ".", "·")))
```
Comment (NicholasBlaskey) [inline master/internal/trials/postgres_trials.go:75]: blocking: ```db.ApplyPolymorphicFilter``` needs to be updated to have this escaping too right?

File: master/internal/trials/postgres_trials.go [hunk 3]
```diff
@@ -70,8 +71,8 @@ func MetricsTimeSeries(trialID int32, startTime time.Time,
 		case db.MetricTypeBool:
 			cast = "boolean"
 		}
-		subq = subq.ColumnExpr("(metrics->?->>?)::? as ?",
-			metricsObjectName, metricName, bun.Safe(cast), bun.Ident(metricName))
+		subq = subq.ColumnExpr("(metrics->?->>?)::? as ?", metricsObjectName,
```
Comment (NicholasBlaskey) [inline master/internal/trials/postgres_trials.go:74]: the bug is just with ```bun.Ident``` and ```.``` right? Just wondering if there is any where else in the code base that might also have this

**Review by NicholasBlaskey — COMMENTED** [review]
Looks pretty good one blocking comment

File: master/internal/trials/postgres_trials.go (hunk 2 above)
Comment (mapmeld) [reply master/internal/trials/postgres_trials.go:75]: ✅ agreed

File: master/internal/trials/postgres_trials.go (hunk 3 above)
Comment (mapmeld) [reply master/internal/trials/postgres_trials.go:74]: Yes it's `bun.Ident`. We only use `bun.Ident` here and ExperimentIdFilter (which shouldn't be included for separate reasons; it is not supported by proto; see http://latest-main.determined.ai:8080/api/v1/experiments?experimentIdFilter={%22incl%22:[2841]} )

File: master/internal/trials/postgres_trials.go (hunk 2 above)
Comment (hamidzr) [inline master/internal/trials/postgres_trials.go:75]: question: Is there a way for us to escape the problematic char instead of replacing it?

**Review by hamidzr — COMMENTED** [review]
Thanks for working on this. The tests would also be a great addition.

File: master/internal/trials/postgres_trials.go (hunk 2 above)
Comment (mapmeld) [reply master/internal/trials/postgres_trials.go:75]: I don't believe so, because `bun.Ident` is going to do something with it no matter what, and I don't want to lose or reimplement any other escaping done by `bun.Ident`

File: master/internal/trials/postgres_trials.go [hunk 4]
```diff
@@ -98,17 +99,17 @@ func MetricsTimeSeries(trialID int32, startTime time.Time,
 		return metricMeasurements, errors.Wrapf(err, "failed to get metrics to sample for experiment")
 	}
 
-	selectMetrics := map[string]bool{}
+	selectMetrics := map[string]string{}
 
 	for i := range metricNames {
-		selectMetrics[metricNames[i]] = true
+		selectMetrics[strings.ReplaceAll(metricNames[i], ".", "·")] = metricNames[i]
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply master/internal/trials/postgres_trials.go:105]: assuming we're okay with the limitation which sounds reasonable: should we error out here in this case instead or log the situation??

**Review by hamidzr — APPROVED** [review]

File: master/internal/trials/postgres_trials.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mapmeld) [reply master/internal/trials/postgres_trials.go:105]: I think it's such a rare edge case for someone to have metric names only differentiated with "." and "·"; the middle dot is somewhat rare
